src/common/module_builder.cc

```diff
@@ -709,4 +718,115 @@ std::optional<mlir::sdy::MeshOp> ModuleBuilder::getFirstShardyMeshOp(
   return mesh_op;
 }
 
+// MLIR Dumping Helper Functions
+std::string ModuleBuilder::getModelName() {
+  const char *model_name_env = std::getenv("MODEL_NAME");
+  if (model_name_env) {
+    LOG_F(INFO, "Model name from environment: %s", model_name_env);
+    std::string model_name = std::string(model_name_env);
+
+    std::string cleaned_name;
+    cleaned_name.reserve(std::string(model_name).size());
+    for (char c : model_name) {
+      if (std::isalnum(c)) {
+        // Keep alphanumeric characters, convert to lowercase
+        cleaned_name += std::tolower(c);
+      } else if (c == '_' || c == '-') {
+        // Keep underscores and hyphens as they're common in model names
+        cleaned_name += c;
+      }
+    }
+    return cleaned_name;
+  }
+  LOG_F(INFO,
+        "MODEL_NAME environment variable is not set. Using default name.");
+  return "default_model"; // Provide a default name if the env var isn't set
```

Comment:
LLVM/MLIR coding style: Use early exits to simplify the code https://llvm.org/docs/CodingStandards.html#id40
```suggestion
  if (!model_name_env) {
    LOG_F(INFO,
        "MODEL_NAME environment variable is not set. Using default name.");
    return "default_model"; // Provide a default name if the env var isn't set
  }

  LOG_F(INFO, "Model name from environment: %s", model_name_env);
  std::string model_name = std::string(model_name_env);

  std::string cleaned_name;
  cleaned_name.reserve(std::string(model_name).size());
  for (char c : model_name) {
    if (std::isalnum(c)) {
      // Keep alphanumeric characters, convert to lowercase
      cleaned_name += std::tolower(c);
    } else if (c == '_' || c == '-') {
      // Keep underscores and hyphens as they're common in model names
      cleaned_name += c;
    }
  }
  return cleaned_name;
```